ee-feature-pack/galleon-common/src/main/resources/layers/standalone/webservices/layer-spec.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="webservices">
+    <dependencies>
+        <layer name="web-server"/>
+        <layer name="ejb" optional="true"/>
```

Comment:
Is remote EJB support the common case with webservices? If so this is ok; otherwise ejb-lite is better as it has fewer dependencies. If a user wanted @Remote they could just add the 'ejb' layer and have that.

I see a few tests in testsuite/integration/ws use @Remote EJBs but if that's just for test coverage and it's not the typical use case those could be run in a separate execution with a server that included the 'ejb' layer.

Sorry if this has been discussed and I missed it or forgot.

Reply:
I tried with ejb-lite and there are a lot of missing remote failures from testsuite/integration/ws module like : 
```
org.jboss.arquillian.container.spi.client.container.DeploymentException: Cannot deploy stateless-ws-endpoint-example.jar: 
{"WFLYCTL0062: Composite operation failed and was rolled back. Steps that failed:" => {"Operation step-1" => 
{"WFLYCTL0412: Required services that are not installed:" => ["org.wildfly.ejb.remote","org.wildfly.transactions.remote-
transaction-service"],"WFLYCTL0180: Services with missing/unavailable dependencies" => 
["jboss.deployment.unit.\"stateless-ws-endpoint-
example.jar\".component.SimpleStatelessWebserviceEndpointImpl.CREATE is missing [org.wildfly.transactions.remote-
transaction-service]","jboss.deployment.unit.\"stateless-ws-endpoint-
example.jar\".component.SimpleStatelessWebserviceEndpointImpl.VIEW.\"org.jboss.as.test.integration.ws.ejb.SimpleState
lessWebserviceEndpointIface\".REMOTE is missing [org.wildfly.ejb.remote]"]}}}

```

Reply:
When I tried, 5 out of 58 tests failed.

I think those are the tests where the EJBs are annotated with @Remote. If so ,they don't pass for a valid reason; they'd be testing something outside the intended use case of the layer. But the presence of those tests doesn't mean that's the use case our layer should target. They could be excluded or tested in a separate execution with the 'ejb' layer added, as you did with the tests that required legacy security. The layer should be designed to be as slim as possible while still meeting the expected end user use cases.

We wouldn't want to base on ejb-lite and make the user provision both 'webservices' and 'ejb' instead of just 'webservices' if wanting remote EJB is the common use case. On the other hand, 'ejb' brings the remoting subsytem and an endpoint, which adds both weight and security risk. If a user didn't want that they'd have to use 'webservices' and 'ejb-lite' and exclude 'ejb', which is fairly painful and not intuitive. It's doable though, so if wanting remote EJBs is the common use case you're targeting it's ok.

Reply:
@bstansberry  Looked at layer dependency twice, and I agree with you that we should make sure the layer to be as slim as possible. I added the "ejb" layer dependency as user can get a out of box webservices layer which works for most of the causes, but this will increase the installation size and make the layer exclusion painful for user. I didn't think of the scenario if user wants to exclude something.  I pushed the change to change the dependency to "ejb-lite" as you suggested.